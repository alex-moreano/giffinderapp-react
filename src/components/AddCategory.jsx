import {useState} from 'react';

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    
    const onInputChange = ({target})=>{
        setInputValue(target.value);
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        let newInputValue = inputValue.trim();
        if(newInputValue.length <= 1) return;
        newInputValue = newInputValue;
        onNewCategory(newInputValue);
        setInputValue('');
    }

  return (
    <form onSubmit={(event)=>onSubmit(event)}>
        <input
        type="text"
        placeholder="Search Gif..."
        value={inputValue}
        onChange={onInputChange}
        />
    </form>
  )
  AddCategory.propTypes={
    onNewCategory: PropTypes.func.isRequired
  }
}
