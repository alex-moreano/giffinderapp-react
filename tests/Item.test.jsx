import { render } from "@testing-library/react";
import { GifGridItem } from "../src/components/GifGridItem";

describe('Tests on gif grid item component',()=>{
    test('Title and url must be on the props', () => {
        const title = 'This is a title';
        const {getByText} = render (<GifGridItem title={title}/>);
        expect(getByText(title)).toBeTruthy();
    });

    /*test('Url must be on the props and title on alt', () => {
        render (<GifGridItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });*/

    test('Must match the snapshot', () => {
        const title = 'This is a title';
        const url = 'https://www.something.com';
        const {container} = render (<GifGridItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });
});