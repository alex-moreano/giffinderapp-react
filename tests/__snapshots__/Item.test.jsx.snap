// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on gif grid item component Must match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is a title"
      src="https://www.something.com"
    />
    <p>
      This is a title
    </p>
  </div>
</div>
`;
